src/server/analytic_services/history_data_store.js

```diff
@@ -34,36 +33,33 @@ class HistoryDataStore {
             system_snapshot: item,
             history_type: 'SYSTEM'
         };
-        return P.resolve()
-            .then(() => this._history.validate(record))
-            .then(() => this._history.col().insertOne(record))
-            .then(() => this._history.col().removeMany({
-                // remove old snapshots
-                time_stamp: { $lt: record_expiration_date },
-                history_type: 'SYSTEM'
-            }));
+        this._history.validate(record);
+        await this._history.insertOne(record);
+        await this._history.deleteMany({
+            // remove old snapshots
+            time_stamp: { $lt: record_expiration_date },
+            history_type: 'SYSTEM'
+        });
     }
 
-    get_pool_history() {
-        return this._history.col().find({
-                history_type: 'SYSTEM'
-            }, {
+    async get_pool_history() {
+        return this._history.find({
+            history_type: 'SYSTEM'
+        }, {
+            projection: {
                 time_stamp: 1,
                 'system_snapshot.pools': 1,
-            })
-            .toArray();
+            }
+        });
     }
 
-    get_system_version_history() {
-        return P.resolve()
-            .then(() => this._history.col().find({
-                    history_type: 'VERSION'
-                }, {
-                    time_stamp: 1,
-                    version_snapshot: 1,
-                })
-                .toArray()
-            );
+    async get_system_version_history() {
+        return this._history.find({
+            history_type: 'VERSION'
+        }, {
+            time_stamp: 1,
```

Comment:
projection missing

Reply:
Somehow missed it thanks.
Browsed all files with .find( and .findOne(
Did not find any occurrences.